src/plugins/data_source/server/routes/test_connection.ts

```diff
@@ -53,6 +53,20 @@ export const registerTestConnectionRoute = async (
                     ]),
                   }),
                 }),
+                schema.object({
+                  type: schema.string({
+                    validate: (value) => {
+                      if (
+                        value === AuthType.NoAuth ||
+                        value === AuthType.UsernamePasswordType ||
+                        value === AuthType.SigV4
+                      ) {
+                        return `Must not be no_auth or username_password or sigv4 for registered auth types`;
```

Comment:
Will this show up in the response error message? Can we add a test for this?

Reply:
Thanks for the comment.

Test case updated here:
https://github.com/opensearch-project/OpenSearch-Dashboards/pull/6109/files#diff-5437e0f1483390165b21a3beaeccae1092205fb9d90de5a95e4d792e4826966fR260

Thanks